Remove separate altitude parameter from Astro binding

When approving my auto-discovered Sun thing, I got this in my log:

```
15:52:30.491 [INFO ] [smarthome.event.ThingAddedEvent     ] - Thing 'astro:sun:local' has been added.
15:52:30.493 [INFO ] [me.event.ThingStatusInfoChangedEvent] - 'astro:sun:local' changed from UNINITIALIZED to INITIALIZING
15:52:30.500 [ERROR] [ding.astro.handler.AstroThingHandler] - Astro parameters geolocation could not be split into latitude and longitude, disabling thing 'astro:sun:local'
15:52:30.501 [INFO ] [me.event.ThingStatusInfoChangedEvent] - 'astro:sun:local' changed from INITIALIZING to OFFLINE
```
The issue is that the discovery now adds "<lat>,<long>,<alt>" in the geoLocation config parameter of the Thing, while it expects the altitude to be filled into the dedicated "altitude" parameter instead.

As we will soon have a [location context](https://github.com/eclipse/smarthome/issues/3709), which also includes altitude information, I think it would make most sense to remove the "altitude" parameter of Astro things and rather mark the geoLocation parameter to be of context "location", which implicitly can hold an optional altitude value.

--- a/extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/config/config.xml
+++ b/extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/config/config.xml
@@ -6,12 +6,9 @@
 
 	<config-description uri="thing-type:astro:config">
 		<parameter name="geolocation" type="text" required="true">
+			<context>location</context>
 			<label>GeoLocation</label>
-			<description>The latitude and longitude separated with a comma (xx.xxxxxx,xx.xxxxxx)</description>
-		</parameter>
-		<parameter name="altitude" type="integer" required="false">
-			<label>Altitude</label>
-			<description>The altitude of the GeoLocation in meter above sea level.</description>
+			<description>The latitude, longitude and altitude separated with a comma (xx.xxxxxx,xx.xxxxxx,xx)</description>
 		</parameter>
 		<parameter name="interval" type="integer" min="1" max="86400">
 			<label>Interval</label>

--- a/extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/config/config.xml
+++ b/extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/config/config.xml
@@ -7,8 +7,8 @@
 	<config-description uri="thing-type:astro:config">
 		<parameter name="geolocation" type="text" required="true">
 			<context>location</context>
-			<label>GeoLocation</label>
-			<description>The latitude, longitude and altitude separated with a comma (xx.xxxxxx,xx.xxxxxx,xx)</description>
+			<label>Location</label>
+			<description>The latitude, longitude and altitude separated with a comma (<lat>,<long>,<alt>)</description>
 		</parameter>
 		<parameter name="interval" type="integer" min="1" max="86400">
 			<label>Interval</label>

--- a/extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/calc/SunCalc.java
+++ b/extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/calc/SunCalc.java
@@ -57,7 +57,7 @@ public class SunCalc {
     /**
      * Calculates the sun position (azimuth and elevation).
      */
-    public void setPositionalInfo(Calendar calendar, double latitude, double longitude, Integer altitude, Sun sun) {
+    public void setPositionalInfo(Calendar calendar, double latitude, double longitude, Double altitude, Sun sun) {
         double lw = -longitude * DEG2RAD;
         double phi = latitude * DEG2RAD;
 
@@ -82,7 +82,7 @@ public class SunCalc {
     /**
      * Calculates sun radiation data.
      */
-    public void setRadiationInfo(Calendar calendar, double elevation, Integer altitude, Sun sun) {
+    public void setRadiationInfo(Calendar calendar, double elevation, Double altitude, Sun sun) {
         double sinAlpha = Math.sin(DEG2RAD * elevation);
 
         int dayOfYear = calendar.get(Calendar.DAY_OF_YEAR);
@@ -111,7 +111,7 @@ public class SunCalc {
     /**
      * Returns true, if the sun is up all day (no rise and set).
      */
-    private boolean isSunUpAllDay(Calendar calendar, double latitude, double longitude, Integer altitude) {
+    private boolean isSunUpAllDay(Calendar calendar, double latitude, double longitude, Double altitude) {
         Calendar cal = DateTimeUtils.truncateToMidnight(calendar);
         Sun sun = new Sun();
         for (int minutes = 0; minutes <= MINUTES_PER_DAY; minutes += CURVE_TIME_INTERVAL) {
@@ -127,11 +127,11 @@ public class SunCalc {
     /**
      * Calculates all sun rise and sets at the specified coordinates.
      */
-    public Sun getSunInfo(Calendar calendar, double latitude, double longitude, Integer altitude) {
+    public Sun getSunInfo(Calendar calendar, double latitude, double longitude, Double altitude) {
         return getSunInfo(calendar, latitude, longitude, altitude, false);
     }
 
-    private Sun getSunInfo(Calendar calendar, double latitude, double longitude, Integer altitude, boolean onlyAstro) {
+    private Sun getSunInfo(Calendar calendar, double latitude, double longitude, Double altitude, boolean onlyAstro) {
         double lw = -longitude * DEG2RAD;
         double phi = latitude * DEG2RAD;
         double j = DateTimeUtils.midnightDateToJulianDate(calendar) + 0.5;

--- a/extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/config/AstroThingConfig.java
+++ b/extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/config/AstroThingConfig.java
@@ -22,7 +22,7 @@ import org.apache.commons.lang.builder.ToStringStyle;
  */
 public class AstroThingConfig {
     private String geolocation;
-    private Integer altitude;
+    private Double altitude;
     private Double latitude;
     private Double longitude;
     private Integer interval;
@@ -39,7 +39,7 @@ public class AstroThingConfig {
         } else if (geoParts.length == 3) {
             latitude = toDouble(geoParts[0]);
             longitude = toDouble(geoParts[1]);
-            altitude = Integer.parseInt(geoParts[2]);
+            altitude = toDouble(geoParts[2]);
         }
     }
 
@@ -75,7 +75,7 @@ public class AstroThingConfig {
     /**
      * Returns the longitude.
      */
-    public Integer getAltitude() {
+    public Double getAltitude() {
         return altitude;
     }
 

--- a/extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/config/AstroThingConfig.java
+++ b/extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/config/AstroThingConfig.java
@@ -36,6 +36,10 @@ public class AstroThingConfig {
         if (geoParts.length == 2) {
             latitude = toDouble(geoParts[0]);
             longitude = toDouble(geoParts[1]);
+        } else if (geoParts.length == 3) {
+            latitude = toDouble(geoParts[0]);
+            longitude = toDouble(geoParts[1]);
+            altitude = Integer.parseInt(geoParts[2]);
         }
     }
 

